src/nvim/option.c

```diff
@@ -4195,6 +4255,21 @@ set_num_option (
 
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting && errmsg == NULL) {
+    char_u buf_old[11], buf_new[11], buf_type[7];
+    vim_snprintf((char *)buf_old, 10, "%ld", old_value);
```

Comment:
Do we have negative values for options (`-1234567890` would need a `12` char buffer) or option values bigger than `INT_MAX` (`old_value` has type `long`, could need a >11 char buffer)?